windows/devices/uwb/include/windows/devices/uwb/UwbDeviceConnector.hxx

```diff
@@ -48,14 +50,31 @@ public:
      * @return false If listening for notifications could not be started.
      */
     bool
-    NotificationListenerStart(std::function<void(::uwb::protocol::fira::UwbNotificationData)> onNotification);
+    NotificationListenerStart();
 
     /**
      * @brief Stop listening for notifications.
      */
     void
     NotificationListenerStop();
 
+    /**
+     * @brief Registers the callbacks for a particular session
+     *
+     * @param sessionId
+     * @param callbacks
+     */
+    void
+    RegisterSessionEventCallbacks(uint32_t sessionId, std::weak_ptr<::uwb::UwbSessionEventCallbacks> callbacks);
+
+    /**
+     * @brief Sets the callbacks for the UwbDevice that owns this UwbDeviceConnector
+     *
+     * @param callbacks
+     */
+    void
+    RegisterDeviceEventCallbacks(::uwb::UwbDeviceEventCallbacks callbacks);

```

Comment:
One thing missing here is the ability to unregister the callbacks. When a `weak_ptr` is passed, the un-registration can be implicit since the caller can simply delete the object containing the callbacks and the connector then won't be able to resolve them and can remove its copy of the `weak_ptr`. Even in that case, though, it would be better to have a solution that allowed explicit deregistration to support the scenario where the callbacks are provided in a class that implements multiple interfaces simultaneously, eg.:

```
struct UberUwbSessionImpl : 
    public UwbDeviceEventCallbacks, public UwbSessionEventCallbacks, public ISomeOtherInterface
{
    // impl here
};
```

One way to allow un-registration is to return some kind of opaque token that the implementation uses to associate the callbacks with the registration, then the caller uses the token to un-register its callbacks. This can be done with some kind of unique identifier, an opaque pointer, or some combination thereof.

Reply:
I added a skeleton for deregistration. I think it makes the most sense for session callbacks, as sessions come and go but there is only one device that owns a UwbDeviceConnector so we might not need it for device callbacks. We can figure more details out in the future.